gpt4all-chat/chatllm.cpp

```diff
@@ -261,6 +261,10 @@ bool ChatLLM::loadModel(const ModelInfo &modelInfo)
             m_llModelInfo.model = LLModel::Implementation::construct(filePath.toStdString(), buildVariant, n_ctx);
 
             if (m_llModelInfo.model) {
+                if (m_llModelInfo.model->isModelBlacklisted(filePath.toStdString())) {
+                    // TODO(cebtenzzre): warn that this model is out-of-date
```

Comment:
@manyoso I'm not sure how to best display a warning to the user here. Maybe you could implement this?
Note: isModelBlacklisted will synchronously read and hash the entire file if the filename is on the list (currently only one).

Reply:
The normal way is to admit a signal and then catch that signal in chat.cpp/h class and propagate to GUI which is in same thread as chat.cpp/h